### ISSUE TITLE
IE search strategy no longer returned when OS_DEFAULT strategy used on Windows

### ISSUE THREAD

**Issue by jonpasski** [MarkusBernhardt/proxy-vole#30]
In proxy-vole 1.0.3, the IE search strategy is added when the OS_DEFAULT search strategy is used on Windows. In proxy-vole 1.0.4, the strategy is not added when OS_DEFAULT is used. Is this expected behavior?

**Comment by nerdydrew** [comment]
That's correct. In 1.0.3 and earlier, the `WinProxySearchStrategy` just replicated the `IEProxySearchStrategy`. The original author left a TODO back in 2009 to implement it with Windows API calls. I did so in https://github.com/MarkusBernhardt/proxy-vole/commit/62b129a79c6e58f040c93855f6f977290fa8ff32, which was included in 1.0.4. Now, you can use the Windows strategy to find proxy settings in the registry and the IE strategy for proxy settings at the user level like before.

**Comment by MarkusBernhardt** [comment]
Shouldn't we have in Win and IE Stratege as OS_DEFAULT?

Von meinem iPhone gesendet

**Comment by jonpasski** [comment]
Thanks for the background @nerdydrew. As a recent consumer of the library, I was surprised by the change. I could see others who have been using this, or the previously maintained, version of the library surprised also. This is especially true if it has been the default for the last 7-8 years. Since the above change will break those existing consumers who do not explicitly add the IE strategy and who upgrade to 1.0.4, I recommend documenting the change more prominently and perhaps bumping the version number to 1.1.

@MarkusBernhardt I'm unfamiliar with Windows, however, I could see the usefulness of allowing users to choose exclusively between the registry settings or the IE settings. I'm using the library with an intercepting proxy tool called [Burp](https://portswigger.net/burp) that's used in web application security assessments. I could see it being useful to reconfigure IE to go through Burp only leaving the desktop settings in place.
